src/System.Linq.Parallel/src/System/Linq/Parallel/Helpers.cs

```diff
@@ -133,7 +134,7 @@ internal struct Slot
         {
             internal int hashCode;
             internal int next;
-            internal TElement value;
+            [AllowNull] internal TElement value;
```

Comment:
💭 It seems like `MaybeNull` should always be applied to a field which has `AllowNull`.

Reply:
Yes, seems fields should be `MaybeNull`, arguments mostly could be `AllowNull`. But compiler complaining for setting null or default value into `MaybeNull` field, but fine with `AllowNull` field, so instead of adding many bangs `!` I used `AllowNull`  for internal class internal methods which are not affecting public APIs, it was appropriate for calling route and called route to avoid many bangs `!`, but of course if still `MaybeNull` is preferred I would update